quantipy/core/tools/dp/prep.py

```diff
@@ -1093,21 +1093,65 @@ def merge_column_metadata(left_column, right_column, overwrite=False):
     """
     Merge the metadata from the right column into the left column.
     """
+    _compatible_types(left_column, right_column)
     left_column['text'] = merge_text_meta(
             left_column['text'],
             right_column['text'],
             overwrite=overwrite)
-    try:
-        if 'values' in left_column:
-            left_column['values'] = merge_values_meta(
-                left_column['values'],
-                right_column['values'],
-                overwrite=overwrite)
-    except KeyError:
-        msg = "Found 'values' object in left {}, but not in right dataset!"
-        print msg.format(left_column['name'])
+    if 'values' in left_column and 'values' in right_column:
+        left_column['values'] = merge_values_meta(
+            left_column['values'],
+            right_column['values'],
+            overwrite=overwrite)
     return left_column
 
+def _compatible_types(left_column, right_column):
+    l_type = left_column['type']
+    r_type = right_column['type']
+    if l_type == r_type: return None
+    all_types = ['array', 'int', 'float', 'single', 'delimited set', 'string',
+                 'date', 'time', 'boolean']
+    err = {
+        'array': all_types,
+        'int': [
+            'float', 'delimited set', 'string', 'date', 'time', 'array'],
+        'float': [
+            'delimited set', 'string', 'date', 'time', 'array'],
+        'single': [
+            'delimited set', 'string', 'date', 'time', 'array'],
+        'delimited set': [
+            'string', 'date', 'time', 'array'],
+        'string': [
+            'int', 'float', 'single', 'delimited set', 'date', 'time', 'array'],
+        'date': [
+            'int', 'float', 'single', 'delimited set', 'string', 'time', 'array'],
+        'time': [
+            'int', 'float', 'single', 'delimited set', 'string', 'time', 'array'],
+        }
+    warn = {
+        'int': [
+            'single'],
+        'float': [
+            'int', 'single'],
+        'single': [
+            'int', 'float'],
+        'delimited set': [
+            'single', 'int', 'float'],
```

Comment:
`int` should only be allowed if there is metadata validation
`float` should err not warn